httpbp/trust_handers.go

```diff
@@ -0,0 +1,233 @@
+package httpbp
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/signing"
+)
+
+const (
+	// EdgeContextSignatureHeader is the key use to get the signature for
+	// the edge context headers from the HTTP request headers.
+	EdgeContextSignatureHeader = "X-Edge-Request-Signature"
+
+	// SpanSignatureHeader is the key use to get the signature for
+	// the span headers from the HTTP request headers.
+	SpanSignatureHeader = "X-Span-Signature"
+)
+
+// HeaderTrustHandler provides an interface PopulateBaseplateRequestContext to
+// verify that it should trust the HTTP headers it receives.
+type HeaderTrustHandler interface {
+	// TrustEdgeContext informs the function returned by PopulateBaseplateRequestContext
+	// if it can trust the HTTP headers that can be used to create an edge
+	// context.
+	//
+	// If it can trust those headers, then the headers will be copied into the
+	// context object to be later used to initialize the edge context for the
+	// request.
+	TrustEdgeContext(r *http.Request) bool
+
+	// TrustSpan informs the function returned by PopulateBaseplateRequestContext
+	// if it can trust the HTTP headers that can be used to create a server
+	// span.
+	//
+	// If it can trust those headers, then the headers will be copied into the
+	// context object to later be used to initialize the server span for the
+	// request.
+	TrustSpan(r *http.Request) bool
+}
+
+// NeverTrustHeaders implements the HeaderTrustHandler interface and always
+// returns false.
+type NeverTrustHeaders struct{}
```

Comment:
Is there a use for those beyond tests? If not I would move them into the test.

Reply:
Historically this is what most HTTP services used 😅 , if anything `AlwaysTrust` is the one to remove.

Reply:
For users using different signatures than ours? I think these 2 makes sense to provide as part of your API as they provide useful and reasonable alternatives to the users.

Reply:
The way I would think about it is:

1. `NeverTrustHeaders` you are exposed to the public internet and also do not expect to receive these headers anyways.
2. `AlwaysTrustHeaders` you only accept calls from within a trusted network and you feel comfortable always trusting these headers.
3. `HeaderSignatureTrustHandler` you are somewhere in between.  Maybe you are exposed to the public internet but also receive calls from internal clients and you want to be able to trust these headers conditionally.

Reply:
(I'm adding those blurbs to the documentation for the appropriate trust handlers.)